pay-api/src/pay_api/models/transactions_materialized_view.py

```diff
@@ -0,0 +1,119 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""transactions_materialized_view materialized view."""
+
+from pay_api.models import db
+
+
+class TransactionsMaterializedView(db.Model):
+    """
+    This model represents the `transactions_materialized_view` materialized view in the database.
+
+    It is designed to improve query performance for purchase history and other related queries by
+    pre-joining data from multiple tables, thereby reducing the need for complex joins in real-time queries.
+    SELECT ... FROM invoices
+    LEFT OUTER JOIN payment_accounts ON invoices.payment_account_id = payment_accounts.id
+    LEFT OUTER JOIN payment_line_items ON payment_line_items.invoice_id = invoices.id
+    LEFT OUTER JOIN fee_schedules ON fee_schedules.fee_schedule_id = payment_line_items.fee_schedule_id
+    LEFT OUTER JOIN invoice_references ON invoice_references.invoice_id = invoices.id
+    ORDER BY invoices.id DESC;
+
+    Note:
+    - This model should be treated as read-only.
+    Any updates to the underlying tables will not automatically reflect in this materialized view
+    until it is refreshed.
+    - Use this model to query data in the materialized view rather than directly
+    joining multiple tables when possible for performance benefits.
+    """
+
+    __tablename__ = 'transactions_materialized_view'
+    __table_args__ = {'extend_existing': True}
+    __mapper_args__ = {
+        "include_properties": [
+            "auth_account_id",
+            "bcol_account",
+            "billable",
+            "business_identifier",
+            "corp_type_code",
+            "created_by",
+            "created_name"
+            "created_on",
+            "description",
+            "details",
+            "disbursement_date",
+            "disbursement_reversal_date",
+            "fee_schedule_id",
+            "filing_id",
+            "filing_type_code",
+            "folio_number",
+            "gst",
+            "invoice_id",
+            "invoice_number",
+            "invoice_reference_id",
+            "invoice_reference_status_code",
+            "invoice_status_code",
+            "line_item_id",
+            "overdue_date",
+            "paid",
+            "payment_account_id",
+            "payment_account_name",
+            "payment_date",
+            "payment_method_code",
+            "pst",
+            "reference_number",
+            "refund_date",
+            "refund",
+            "row_id",
+            "service_fees",
+            "total",
+        ]
+    }
+
+    row_id = db.Column(db.Integer, primary_key=True)
+
+    auth_account_id = db.Column(db.String(50), nullable=True)
```

Comment:
I don't think we need all of these, because they'd be duplicated now so not a good idea

Reply:
What if we want to change a field in here and invoices? we're going to have to change it in 2 spots.. which isn't good